espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/SendKeys.java

```diff
@@ -65,6 +68,22 @@ public Void handle(TextParams params) throws AppiumException {
             viewInteraction.perform(typeText(value));
         } catch (PerformException e) {
             throw new InvalidElementStateException("sendKeys", params.getElementId(), e);
+        } catch (RuntimeException e) {
+            if (!e.getMessage().contains("IME does not understand how to translate")) {
+                throw e;
```

Comment:
the way i am reading this control flow, the only way we will try replaceText is if we get a `RuntimeException`  with the content "IME does not understand how to translate". Is that what's intended? Or did we want to try replaceText in other failure scenarios as well?

Reply:
yes, we only want to use replaceText for this particular case